lib/channel_model.js

```diff
@@ -74,10 +98,16 @@ C.open = function() {
 
 C.close = function() {
   var self = this;
-  return Promise.fromCallback(function(cb) {
+
+  return Bluebird.fromCallback(function(cb) {
     return self.closeBecause("Goodbye", defs.constants.REPLY_SUCCESS,
                     cb);
   });
+  // return new Promise(function(resolve, reject) {
```

Comment:
Should this be commented out?

Reply:
I need some help with this.

If I replace `Bluebird.fromCallback` with commented code it seems that `channel.accept` is called _after_ this promise resolved. So the behavior changes and I have no idea why. I'm sure you have more context with this.